app/jobs/attach_remote_file_to_upload_job.rb

```diff
@@ -6,11 +6,26 @@ class AttachRemoteFileToUploadJob < ApplicationJob
   queue_as :default
 
   def perform(upload)
-    upload.files.attach(io: URI.open(upload.url), filename: upload.name)
+    io = URI.open(upload.url)
+    upload.files.attach(io: io, filename: filename_from_io(io) || upload.name)
   rescue SocketError, OpenURI::HTTPError => e
     error = "Error opening #{url}: #{e}"
     Rails.logger.info(error)
     Honeybadger.notify(error)
     raise e
   end
+
+  private
+
+  def filename_from_io(io)
```

Comment:
Who woulda thought that parsing HTTP headers in Rails would be... this... interesting..

Reply:
To be fair, I haven't actually tried this with something that has a content-disposition but nothing I've ready leads me to believe rails does anything great for us here (although apparently mechanize does have some niceties around this, but that method is, uh, [something](https://github.com/sparklemotion/mechanize/blob/b959fb10d2e5943e69c268993481a1b73c80717e/lib/mechanize/parser.rb#L106) [else](https://github.com/sparklemotion/mechanize/blob/b959fb10d2e5943e69c268993481a1b73c80717e/lib/mechanize/http/content_disposition_parser.rb))